utils/mgmt_system.py

```diff
@@ -71,6 +77,32 @@ def get_search_filter_spec(self, *args, **kwargs):
 
         return pfs
 
+    def __getattribute__(self, attr):
+        # fetch the attribute using parent class to avoid recursion
+        res = super(_PsphereClient, self).__getattribute__(attr)
+        # any callable (except login) is login protected
```

Comment:
If everything but login is special, doesn't that actually mean that login is special?